week-1-ts-ecosystem/w01e04-light-me-up/src/solution1.ts

```diff
@@ -0,0 +1,51 @@
+/**
+ * W zasadzie tutaj możesz modyfikować wszystko poza wartościami:
+ *
+ * - 100 w przypadku PowerSource.
+ * - 20 w przypadku LightBulb.
+ *
+ * Pamiętaj `PowerSource` ma nie wiedzieć — kto z niego korzysta!
+ * Dodatkowo — rzucać błąd, jeśli zapas mocy się wyczerpie.
+ * */
+
+class PowerSource {
+    private energySupply = 100
+
+    consume(energy:number) {
+        this.notEnoughEnergyException()
+        this.energySupply -= energy
+    }
+
+    notEnoughEnergyException(){
+        if(this.energySupply === 0){
```

Comment:
Warunek jest błędny - jeśli `this.energySupply` jest na minusie - a taka sytuacja może wystąpić w Twoim kodzie.